### PR TITLE
Remove version check after backport 

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR removes BWC version for the new field `ModelPlotConfig.annotations_enabled`.
The field was introduced in PR https://github.com/elastic/elasticsearch/pull/57539 which has already been backported to 7.9

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/ModelPlotConfig.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public ModelPlotConfig(boolean enabled, String terms, Boolean annotationsEnabled
     public ModelPlotConfig(StreamInput in) throws IOException {
         enabled = in.readBoolean();
         terms = in.readOptionalString();
-        if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
+        if (in.getVersion().onOrAfter(Version.V_7_9_0)) {
```
Comment (droberts195) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/ModelPlotConfig.java:59]: I would just delete the version check from master.  The logic is very simple and unlikely to cause clashes that are hard to resolve even in the unlikely event of someone adding another field in the lifetime of 7.x.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/ModelPlotConfig.java [hunk 2]
```diff
@@ -67,7 +67,7 @@ public ModelPlotConfig(StreamInput in) throws IOException {
     public void writeTo(StreamOutput out) throws IOException {
         out.writeBoolean(enabled);
         out.writeOptionalString(terms);
-        if (out.getVersion().onOrAfter(Version.V_8_0_0)) {
+        if (out.getVersion().onOrAfter(Version.V_7_9_0)) {
```
Comment (droberts195) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/ModelPlotConfig.java:70]: As above, you might as well just delete this line.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/ModelPlotConfig.java (hunk 1 above)
Comment (przemekwitek) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/ModelPlotConfig.java]: Done.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/ModelPlotConfig.java (hunk 2 above)
Comment (przemekwitek) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/ModelPlotConfig.java]: Done.

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**